src/components/ui/highlight-card/highlight-card.type.ts

```diff
@@ -0,0 +1,21 @@
+import { IImage } from '../product-card/product-card.type';
+
+interface IMessageItem {
+  type: 'message';
+  title: string;
+  content: string;
+  highlight: boolean;
+}
+
+interface IProductItem {
+  type: 'product';
+  id: number;
+  price: number;
+  priceBase: number;
+  image: IImage;
+  name: string;
+  specialPrice: number;
```

Comment:
Pode deixar como opcional e mover para o fim da interface
```suggestion
  specialPrice?: number;
```

Reply:
@simaomeyrerdooca ajustado